src/general/GameWorld.cs

```diff
@@ -18,6 +18,15 @@
 [UseThriveSerializer]
 public class GameWorld : ISaveLoadable
 {
+    [JsonProperty]
+    public UnlockProgress UnlockProgress = new();
+
+    [JsonProperty]
+    public int TotalPlayerDeaths;
```

Comment:
When thinking about potential designs for the unlock system, I thought that the unlock progress tracking would be handled by the unlocks system itself because otherwise we end up with a ton of tracking variables in the world or/and the microbe class. So related to my previous comment I think the way to construct the unlock progress would be to create the unlock condition objects once for a started game and then store the state about the unlock conditions in those class instances. Feel free to suggest something else, but do you think a design like that could work?

Reply:
I presumably need to store the total player deaths somewhere within the save file though?

Reply:
Well with my suggestion you'd save the entire tree of unlock condition objects, and each of those would keep track of the statistics they care about. To simplify the design and make it extensible gameplay code should have helper methods made in the main unlock conditions class which then could forward game events to the relevant unlock condition objects (that way gameplay code is less tightly coupled to specific unlock condition implementations).

Reply:
I'm not sure I have time or C# ability to rewrite this PR using that system, so I'll just close it.

It has been interesting playing around with this game and I'd like to thank you for your time and feedback.

Reply:
From my perspective this was almost complete, so it is a bit unfortunate to see that this won't make it all the way to the finish line. But I guess that's how it is. If you want to come back in a couple of months to finish this, you are entirely welcome to do so. Or maybe someone else will be able to complete this feature. 
Anyway I want to also say thanks for at least that one PR you did complete and helped develop Thrive.